### PR TITLE
Admin disable user account for various reasons #187300247

### PR DESCRIPTION
**Pull request by emmanueltct**
#### What does this PR do?
This PR implements the task of admin activate/disable user acount due to various reason and It addresses the requirements outlined in Pivotal ID #187300247 
#### Description of Task to be completed?
Admin should be able to disable an account for various reasons.
when a user is suspected or proved to have violated the terms of service of the application
then admin is able to trigger an action to block usage of a particular account and an email sent to the user with reasons fro the block.
#### How should this be manually tested?
• this task will be test using a swagger documentation
#### What are the relevant pivotal tracker/Trello stories?
https://www.pivotaltracker.com/story/show/187300247

activate 1

activate 2

activate 3

activate 4

### CODE REVIEW

**Review by PrinceRWIGIMBA — COMMENTED** [review]
good

**Review by PrinceRWIGIMBA — APPROVED** [review]

**Review by kayigmb — APPROVED** [review]

File: src/routes/user.routes.ts [hunk 1]
```diff
@@ -63,4 +69,16 @@ userRoutes.patch(
   updateUserPassword
 );
 
+userRoutes.get("/:userId", authenticate, isAdmin, isUserExist, getSingleUser);
+
+userRoutes.patch(
+  "/:userId/status",
+  authenticate,
+  isAdmin,
+  isUserExist,
```
Comment (istafrica-dennw) [inline src/routes/user.routes.ts]: this query is unnecessary. you can know if the user exists when updating

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]
reduce your database querrying if not very necessary

**Review by ErnestTchami — APPROVED** [review]

File: src/routes/user.routes.ts (hunk 1 above)
Comment (emmanueltct) [reply src/routes/user.routes.ts]: I worked on it and removed unnecessary database query

**Review by istafrica-dennw — APPROVED** [review]

**Pull request merged**